### ISSUE TITLE
Is the something wrong when I mix special characters with letters and numbers?

### ISSUE THREAD

**Issue by StraigenDaigen** [explosion/spaCy#5372]
I'm developing a script to extract the entities from financial documents, for example I have been annotating MONEY, PERSON, ORG, ID, etc. All works well when I had trained PERSON and ORG, but when I added MONEY that has ($)character I saw that the performance of the model decrease a lot and it is not detecting PERSON and ORG well... In addition, I have "\n" for line breakes incluided in the dataset. Any suggestion to solve this problem? 

Currently I'm using TagEditor to annotate the texts, my OS is Windows 10 with Anaconda Environment.

**Comment by adrianeboyd** [comment]
Can you show a few examples of your training data (anonymized if necessary) including several entity types?

It's possible that misalignments between the entity spans and spacy's tokenization are leading to problems (there have been a few bugfixes related to this recently), but it'd be easiest to figure out the details given a few concrete examples.

**Comment by StraigenDaigen** [comment]
Hi!, thanks for the help, you can see a sample following: 

> TRAIN_DATA=[
('\r\nSECRETARIA DE EDUCACION MUNICIPAL DE POPAYAN Comprobante de Pago Periodo', {
'entities': [(2, 46, 'SCRT')]
}),

('01-Jan-20', {
'entities': [(0, 9, 'DATE')]
}),
('a 31-Jan-20 Nombres: PARRA GIRALDO', {
'entities': [(2, 11, 'DATE'), (21, 34, 'PERSON')]
}),
('JOSE LUIS Documento:', {
'entities': [(0, 9, 'PERSON')]
}),
('1234567 Ciudad:', {
'entities': [(0, 7, 'ID')]
}),
('Popayan (Cau)', {
'entities': [(0, 7, 'CITY')]
}),
('Centro de Secretaria Municipal De Popayan Dependencia: TALENTO HUMANO Cargo:', {
'entities': [(0, 41, 'CC')]
}),

('$2,242,669 Grado:', {
'entities': [(0, 10, 'MONEY')]
}),

('Ingresos Egresos Totales $2,242,669.00 $542,600.00 SUEBA Sueldo Basico 2,242,669.00', {
'entities': [(25, 38, 'MONEY'), (39, 50, 'MONEY'), (71, 83, 'MONEY')]
}),
('DAVIV 1813 DAVIVIENDA 338,000.00 APEPEN Aporte Empleado Pension Normal 89,800.00 APESDN Aporte Empleado', {
'entities': [(0, 21, 'ORG'), (22, 32, 'MONEY'), (71, 80, 'MONEY')]
}),
('Salud Normal 89,800.00 CLUBB CLUB PROMOTOR BIENESTAR SOCIAL 15,000.00 ERMITA FUNERALES LA', {
'entities': [(13, 22, 'MONEY'), (23, 59, 'ORG'), (60, 69, 'MONEY'), (70, 89, 'ORG')]
}),
('ERMITA', {
'entities': [(0, 6, 'ORG')]
}),
('10,000.00 Totales $2,242,669.00 $542,600.00  ', {
'entities': [(0, 9, 'MONEY'), (18, 31, 'MONEY'), (32, 43, 'MONEY')]
}),

('$1,700,069.00 DIASL:30 Salud:', {
'entities': [(0, 13, 'MONEY')]
}),
('Sanitas S.A. Pension:', {
'entities': [(0, 12, 'ORG')]
}),
('Colpensiones CajaCF:', {
'entities': [(0, 12, 'ORG')]
}),
('Cauca - Comfacauca Cesantias:', {
'entities': [(0, 18, 'ORG')]
}),
('Fondo Nacional De Ahorro Banco:', {
'entities': [(0, 24, 'ORG')]
}),
('BanColombia S.A. 012345678910', {
'entities': [(0, 16, 'ORG')]
})
]

**Comment by adrianeboyd** [comment]
The example data looks totally fine -- no misaligned entity spans. If I train a model on this data and test on the same data (just to check that the model is working by overfitting on a small amount of data) with the `train_ner.py` script, the performance is perfect. The scores excerpt for NER:

```
{'ents_p': 100.0, 'ents_r': 100.0, 'ents_f': 100.0, 'ents_per_type': {
'ORG': {'p': 100.0, 'r': 100.0, 'f': 100.0}, 'MONEY': {'p': 100.0, 'r': 100.0, 'f': 100.0}, 
'SCRT': {'p': 100.0, 'r': 100.0, 'f': 100.0}, 'CITY': {'p': 100.0, 'r': 100.0, 'f': 100.0}, 
'CC': {'p': 100.0, 'r': 100.0, 'f': 100.0}, 'PERSON': {'p': 100.0, 'r': 100.0, 'f': 100.0}, 
'DATE': {'p': 100.0, 'r': 100.0, 'f': 100.0}, 'ID': {'p': 100.0, 'r': 100.0, 'f': 100.0}}}
```

You can check the alignment of your data like this:

```
    from spacy.gold import biluo_tags_from_offsets
    for text, annot in TRAIN_DATA:
        doc = nlp.make_doc(text)
        print(biluo_tags_from_offsets(doc, annot["entities"]))
```

If there are any `-` tags in the output, then some entity offsets didn't align with token boundaries and those annotations will be ignored while training.

**Comment by StraigenDaigen** [comment]
how many annotations I need to do to get good accuracy? Because after 1000 sentences labeled, I got the following results:

![sampla_train](https://user-images.githubusercontent.com/43056610/80753642-37701600-8af3-11ea-8a21-dde3dc2e6aa2.PNG)

As you can see, my loss is too high. I used this configuration.
![spacy_config](https://user-images.githubusercontent.com/43056610/80753765-70a88600-8af3-11ea-8363-741724706dbe.PNG)

**Comment by svlandeg** [comment]
The absolute numeric value of the loss value doesn't really tell you anything by itself. You want to monitor, during training, whether loss goes down. If it does, your model is still learning to fit the training data.

Additionally, you probably want to have some held-out dev test set, on which you measure performance like precision/recall/F-score, to assess whether your models aren't overfitting, and what the actual performance of your models is. An F-score or an accuracy score will tell you much more than some loss value.
